sql/binlog.h

```diff
@@ -879,10 +880,32 @@ class MYSQL_BIN_LOG : public TC_LOG {
     It is called by the threads (e.g. dump thread, applier thread) which want
     to read hot log without LOCK_log protection.
   */
-  my_off_t get_binlog_end_pos() const {
+  my_off_t get_binlog_end_pos(bool need_lock = false,
+                              bool async_slave = false) {
```

Comment:
I think it'll be better to call this parameter wait_for_ack instead of async_slave.

Reply:
done